lib/spack/spack/test/url_fetch.py

```diff
@@ -225,3 +226,28 @@ def test_candidate_urls(pkg_factory, url, urls, version, expected):
     pkg = pkg_factory(url, urls, fetch_options={'timeout': 60})
     f = fs._from_merged_attrs(fs.URLFetchStrategy, pkg, version)
     assert f.extra_options == {'timeout': 60}
+
+
+@pytest.mark.regression('19673')
+def test_missing_curl(tmpdir, monkeypatch):
+    """Ensure a fetch involving missing curl package reports the error."""
+    err_fmt = 'No such command {0}'
+
+    def _which(*args, **kwargs):
+        err_msg = err_fmt.format(args[0])
+        raise spack.util.executable.CommandNotFoundError(err_msg)
+
+    # Patching which_string (versus which) so patch actually works with the
+    # fetch strategy's curl property such that the property is None.
+    monkeypatch.setattr(spack.util.executable, 'which_string', _which)
```

Comment:
```suggestion
    # Patch on fetch_strategy instead of util.executable because it is 
    # imported directly as a symbol
    monkeypatch.setattr(spack.fetch_strategy, 'which', _which)
```

Another alternative would be to change the import in `fetch_strategy.py` to import the module instead of the symbol.

Reply:
True but that would be changing 5 more lines -- to make all of the `which` calls consistent -- than needed for this change.

Reply:
True.  I was trying to avoid changing anything about the module I'm testing -- even imports -- because of previous feedback from Peter.

Reply:
I will make a similar change with a bit more explanation in the comment.